http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeoKubernetesServiceInterp.scala

```diff
@@ -0,0 +1,412 @@
+      appResult <- F.fromOption(appOpt, AppNotFoundException(googleProject, appName, ctx.traceId))
+
+      //TODO implement SAM check
+      hasPermission <- F.pure(true)
+      _ <- if (hasPermission) F.unit else F.raiseError[Unit](AuthorizationError(Some(userInfo.userEmail)))
+
+      canDelete = AppStatus.deletableStatuses.contains(appResult.app.status)
+      _ <- if (canDelete) F.unit
+      else F.raiseError[Unit](AppCannotBeDeletedException(googleProject, appName, appResult.app.status, ctx.traceId))
+//
+      //TODO: do atomically and send message at the same time?
+      _ <- nodepoolQuery.markPendingDeletion(appResult.nodepool.id).transaction
+      _ <- appQuery.markPendingDeletion(appResult.app.id).transaction
+      //TODO queue stuff
+    } yield ()
+
+  private[service] def getSavableCluster(userInfo: UserInfo,
+                                         googleProject: GoogleProject,
+                                         now: Instant): Either[Throwable, SaveKubernetesCluster] = {
+    val auditInfo = AuditInfo(userInfo.userEmail, now, None, now)
+
+    val dummyNodepool = for {
+      nodepoolName <- KubernetesNameUtils.getUniqueName(NodepoolName.apply)
+    } yield Nodepool(
+      NodepoolLeoId(-1),
+      clusterId = KubernetesClusterLeoId(-1),
+      nodepoolName,
+      status = NodepoolStatus.Precreating,
+      auditInfo,
+      machineType = leoKubernetesConfig.nodepoolConfig.dummyNodepoolConfig.machineType,
+      numNodes = leoKubernetesConfig.nodepoolConfig.dummyNodepoolConfig.numNodes,
+      autoscalingEnabled = leoKubernetesConfig.nodepoolConfig.dummyNodepoolConfig.autoscalingEnabled,
+      autoscalingConfig = None,
+      List.empty,
+      List()
+    )
+
+    for {
+      nodepool <- dummyNodepool
+      clusterName <- KubernetesNameUtils.getUniqueName(KubernetesClusterName.apply)
+    } yield SaveKubernetesCluster(
+      googleProject = googleProject,
+      clusterName = clusterName,
+      location = leoKubernetesConfig.clusterConfig.location,
+      status = KubernetesClusterStatus.Precreating,
+      serviceAccount = leoKubernetesConfig.serviceAccountConfig.leoServiceAccountEmail,
+      auditInfo = auditInfo,
+      dummyNodepool = nodepool
+    )
+  }
+
+  private[service] def getDefaultNodepool(clusterId: KubernetesClusterLeoId,
+                                          userInfo: UserInfo,
+                                          req: CreateAppRequest,
+                                          now: Instant): Either[Throwable, Nodepool] = {
+    val auditInfo = AuditInfo(userInfo.userEmail, now, None, now)
+
+    val machineConfig = req.kubernetesRuntimeConfig.getOrElse(
+      KubernetesRuntimeConfig(
+        leoKubernetesConfig.nodepoolConfig.galaxyNodepoolConfig.numNodes,
+        leoKubernetesConfig.nodepoolConfig.galaxyNodepoolConfig.machineType,
+        leoKubernetesConfig.nodepoolConfig.galaxyNodepoolConfig.autoscalingEnabled
```

Comment:
Ah, this function is poorly named from an early iteration. I renamed it to `getUserNodepool`. The dummy nodepool is created from a different config within the `getSavableCluster` function, since saving a cluster explicitly requires the dummy nodepool in the `SaveCluster` ADT